sys/firmware/firmware.c

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright (C) 2017 Kaspar Schleiser
+ *                    Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_firmware
+ * @{
+ *
+ * @file
+ * @brief       Firmware module
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ * @author      Francisco Acosta <francisco.acosta@inria.fr>
+ *
+ * @}
+ */
+
+#include <string.h>
+
+#ifdef RIOT_VERSION
+#include "cpu.h"
+#include "log.h"
+#include "crypto/ed25519_pub.h"
+#include "periph/flashpage.h"
+#else
+#include <stdio.h>
+#define LOG_INFO(...) printf(__VA_ARGS__)
+#endif
+
+#include "firmware.h"
+#include "checksum/fletcher32.h"
+#include "tweetnacl.h"
+
+void firmware_metadata_print(firmware_metadata_t *metadata)
```

Comment:
If this is only for debugging, I think this modification could improve a little the readability of the messages : 

`void firmware_metadata_print(firmware_metadata_t *metadata, uint8_t slot_id)`

Or moving the line 45 (`metadata->start_addr`) at the beginning of this function.